applications/IgaApplication/custom_modelers/nurbs_geometry_modeler_sbm.h

```diff
@@ -0,0 +1,120 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Nicolo' Antonelli
+//                   Andrea Gorgi
+//
+
+# pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "nurbs_geometry_modeler.h"
+
+namespace Kratos {
+
+class KRATOS_API(IGA_APPLICATION) NurbsGeometryModelerSbm
+    : public NurbsGeometryModeler
+{
+public:
+    ///@name Type Definitions
+    ///@{
+    KRATOS_CLASS_POINTER_DEFINITION( NurbsGeometryModelerSbm );
+
+    using IndexType = std::size_t;
+    using SizeType = std::size_t;
+    using NodeType = Node;
+
+    using GeometryType = Geometry<NodeType>;
+    using GeometryPointerType = GeometryType::Pointer;
+
+    using NurbsSurfaceGeometryType = NurbsSurfaceGeometry<3, PointerVector<NodeType>>;
+    using NurbsSurfaceGeometryPointerType = NurbsSurfaceGeometryType::Pointer;
+
+    using NurbsVolumeGeometryType = NurbsVolumeGeometry<PointerVector<NodeType>>;
+    using NurbsVolumeGeometryPointerType = NurbsVolumeGeometryType::Pointer;
+
+    using ContainerNodeType = PointerVector<Node>;
+    using ContainerEmbeddedNodeType = PointerVector<Point>;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    NurbsGeometryModelerSbm()
+        : NurbsGeometryModeler() {}
+
+    /// Constructor.
+    NurbsGeometryModelerSbm(
+        Model & rModel,
+        const Parameters ModelerParameters = Parameters())
+        : NurbsGeometryModeler(rModel, ModelerParameters)
+        , mpModel(&rModel)
+    {
+    }
```

Comment:
I think you should provide some defaults for your SBM stuff. See my comment in the source file.